### PR TITLE
[fixed] IndexLink is active

### PR DESCRIPTION
**Pull request by iNikNik**
IndexLink now being activated correctly. Need to test it with #2079

### CODE REVIEW

**Comment by jayphelps** [comment]
I needed this fix as well, works swimmingly so far and as expected. Fails simply because `pathnameIsActive` is no longer used and should be removed if it doesn't need to be used in alternative unhandled cases.

**Comment by iNikNik** [comment]
@jayphelps unfortunately there is some issues with isActive server rendering ... I'll try to investigate it.
So, should I remove `pathname IsActive`?

**Comment by mjackson** [comment]
Thanks for this PR, @iNikNik! Would you mind rebasing on current master? The tests should pass now.

**Comment by mjackson** [comment]
Oh, also, I believe we've addressed the issues with isActive and server rendering.

**Comment by mjackson** [comment]
I believe this was fixed in 37d9bacd432f1b588df8b1f72227383d1a9f3197

**Comment by iNikNik** [comment]
@mjackson this version didn't fix it. Try to test it with #2079.
https://github.com/rackt/react-router/commit/37d9bacd432f1b588df8b1f72227383d1a9f3197#diff-3b9d7063d56e0abf361937d8f540ff52R82

If pathnamIsActive always return true for index link.

**Comment by iNikNik** [comment]
@mjackson I'll rebase my PR today.

**Comment by mjackson** [comment]
ok, I'll take a closer look.

**Comment by iNikNik** [comment]
@mjackson done!) #2079 tests passed

**Comment by mjackson** [comment]
Thank you @iNikNik! :) Can you please squash down to one commit? I merged the failing tests from #2079 into master.

**Comment by iNikNik** [comment]
@mjackson ok, one moment) Also, I saw you merged tests manually but there was a typo, I'll fix it too

**Comment by iNikNik** [comment]
Oh, god, everything exploded =\

**Comment by mjackson** [comment]
@iNikNik I can do the git work if you'd like :)

**Comment by mjackson** [comment]
Manually rebased and merged in 24e7b4fdf5a80548d55488458cd54c386a688c89. Thanks, @iNikNik! :D

**Comment by iNikNik** [comment]
@mjackson oh, thank you too!) I'm so bad with git ... it's my first PR ever)

File: modules/isActive.js [hunk 1]
```diff
@@ -48,7 +42,7 @@ function routeIsActive(pathname, activeRoutes, activeParams, indexOnly) {
     return false
 
   if (indexOnly)
-    return activeRoutes.length > 1 && activeRoutes[activeRoutes.length - 2].indexRoute === route
+    return activeRoutes.length > 1 && activeRoutes[activeRoutes.length - 1] === route.indexRoute
```
Comment (tomatau) [inline modules/isActive.js]: This is behaving inconsistently for me on a SSR.
Client correctly sets activeRoutes and renders fine.
Server doesn't agree... adding the history as one of the active routes.

server:
```es6
activeRoutes =
   [ { history: undefined, childRoutes: [Object] },
     { path: '/',
       component: [Function: App],
       indexRoute: [Object],
       childRoutes: [Object] },
     { component: [Function: Home] } ]
```

And so the previous code worked, but not this update.

https://github.com/rackt/react-router/issues/1810#issuecomment-147409066

**Pull request closed without merging**